api/controllers/UserSocialMediaUrlController.ts

```diff
@@ -0,0 +1,44 @@
+import { Body, Get, JsonController, Params, Patch, Post, UseBefore } from 'routing-controllers';
+import UserSocialMediaUrlService from '../../services/UserSocialMediaUrlService';
+import { AuthenticatedUser } from '../decorators/AuthenticatedUser';
+import { UserModel } from '../../models/UserModel';
+import { UuidParam } from '../validators/GenericRequests';
+import { GetUserSocialMediaUrlResponse, InsertSocialMediaUrlResponse, UpdateSocialMediaUrlResponse } from '../../types';
+import { UserAuthentication } from '../middleware/UserAuthentication';
+import {
+  InsertSocialMediaUrlRequest,
+  UpdateSocialMediaUrlRequest,
+} from '../validators/UserSocialMediaUrlControllerRequests';
+
+@UseBefore(UserAuthentication)
+@JsonController('/socialMedia')
+export class UserSocialMediaUrlController {
+  private userSocialMediaUrlService: UserSocialMediaUrlService;
+
```

Comment:
I simplified it to just `/user/socialMedia` since I thought that only the user should be allowed to make a `POST`, `PATCH` and `DELETE` request to their own urls. There would not be a scenario where one user is trying to change another user's social media url. Since the `GET` request is now tied to the user, other users are still able to fetch another user's social media urls.

Reply:
Great point, this makes sense to me 👍🏾